data_adapter_oemof/build_datapackage.py

```diff
@@ -384,13 +407,75 @@ def yearly_scalars_to_periodic_values(scalar_dataframe) -> None:
         )
         return scalar_dataframe
 
+    def time_series_aggregation(self, tsam_config: str, destination: str = None):
+        """
+        Aggregates time series in datapackage and saves the new datapackage with updated
+        (sequence)Resources as well as aggregated sequence resources.
+        Parameters
+        ----------
+        tsam_config
+        destination
+
+        Returns
+        -------
+
+        """
+        # Refactor sequences into one Dataframe
+        self.save_datapackage_to_csv(location_to_save_to=destination)
+        sequences = pd.concat(
+            self.parametrized_sequences.values(),
+            axis=1,
+            keys=self.parametrized_sequences.keys(),
+        )
+        sequences.index = self.periods.index
+        sequences["periods"] = self.periods.periods
+
+        # Group sequences by Periods
+        tsam_aggregated_typical_periods = []
+        for period, period_sequence in sequences.groupby(by="periods"):
```

Comment:
This operation might be expensive - I think it could be more performant to simply slice sequences by given length of periods. Aka: 
`tsam.TimeSeriesAggregation(sequences[period_start:period_end], **tsam_config)`
or via `.loc[]`

Reply:
Additionally, we should compare years of resulting chunks with years contained in periods (as we loose this information, when we are not using `groupby`). We could do something like this:
```
if period_sequence[p][0].year != periods[p][0].year:
    raise
if period_sequence[p][-1].year != periods[p][-1].year:
    raise
```

Reply:
Okay as the periods Dataframe is only numeric the groupby should be a lot more perfomrnant here and I could use the found indices for selecting Dataslices? This will also make sure no information is lost while still being easy to read and maintain?

Or I try to select the period indices by something like this which also has to keep the information since the goal of periods was to hold this information. But I cannot estimate how the performance would be, how can I get such information?
```
[tsam.TimeSeriesAggregation(sequences[period_df.index[period_df['periods'] == period], **tsam_config[period]) for period in pd.unique(periods[period)]
```